### PR TITLE
feat: add service account name to scaffold command

### PR DESCRIPTION
**Pull request by Mossaka**
Add a new flag `--service-account-name` to the `spin-plugin-kube scaffold` command
to allow specifying the service account name for the SpinApp CRD.

The default service account name is `default`.

Signed-off-by: Jiaxiao (mossaka) Zhou <duibao55328@gmail.com>

### CODE REVIEW

**Comment by Mossaka** [comment]
@bacongobbler Could you PTAL of this PR?

File: pkg/cmd/scaffold.go [hunk 1]
```diff
@@ -335,6 +341,7 @@ func init() {
 	scaffoldCmd.Flags().StringVar(&scaffoldOpts.memoryLimit, "memory-limit", "", "The maximum amount of memory the application is allowed to use")
 	scaffoldCmd.Flags().StringVar(&scaffoldOpts.memoryRequest, "memory-request", "", "The amount of memory requested by the application. Used to determine which node the application will run on")
 	scaffoldCmd.Flags().BoolVar(&scaffoldOpts.azureWorkloadIdentity, "azure-identity", false, "Enable Azure Workload Identity for the application")
+	scaffoldCmd.Flags().StringVar(&scaffoldOpts.serviceAccountName, "service-account-name", "default", "The name of the service account to use for the application")
```
Comment (kate-goldenring) [inline pkg/cmd/scaffold.go:344]:
```suggestion
	scaffoldCmd.Flags().StringVar(&scaffoldOpts.serviceAccountName, "service-account-name", "", "The name of the service account to use for the application")
```

**Review by kate-goldenring — COMMENTED** [review]
Can we omit the `serviceAccountName` if not provided rather than set it to `default`? I'd rather scaffold a simpler spinapp with only required fields. Unless I missed something and the `serviceAccountName` is now required in the `SpinApp`

**Comment by bacongobbler** [comment]
I have the same review comments as @kate-goldenring. If `--service-account-name` is not specified, it should not be present in the scaffolded output.

I don't see a test that accounts for the default value in the CLI flags. I would add @kate-goldenring's earlier PR suggestion to make the default the empty string, then add a test with `ServiceAccountName` set to the empty string and ensure `serviceAccountName` is not present in the scaffolded manifest output.

File: pkg/cmd/scaffold.go (hunk 1 above)
Comment (Mossaka) [reply pkg/cmd/scaffold.go]: Done

**Comment by Mossaka** [comment]
Thanks for reviewing. I've changed the default value of `--service-account-name` to an empty string and added a test case for it.

**Review by kate-goldenring — APPROVED** [review]
LGTM! Thanks @Mossaka

**Comment by Mossaka** [comment]
Note that this `serviceAccountName` will only work on the main tree of Spin-Operator, not on any released version yet.

**Comment by kate-goldenring** [comment]
Yeah we should release the operator and this and update the compatibility chart
